### ISSUE TITLE
L044 reports a warning when it shouldn't

### ISSUE THREAD

**Issue by barrywhart** [sqlfluff/sqlfluff#2953]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I linted the following file and got an L044 warning.
```
SELECT * FROM (
    SELECT c1 FROM table_2
)
```

### Expected Behaviour

I did not expect this warning, because this query clearly returns 1 result column.

### Observed Behaviour

I did not expect this warning, because this query clearly returns 1 result column.

### How to reproduce

Run `sqlfluff lint` on the query provided above.

### Dialect

ANSI

### Version

0.11.2

### Configuration

n/a

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by barrywhart** [comment]
Oh wow, so embarrassing when I can't reproduce a bug **that I reported**. 🤣 Tested in 0.11.2 and in `main`. Tried a few different dialects. Closing without fix.
